code/web/src/modules/user/Survey.js

```diff
@@ -68,34 +63,74 @@ class Survey extends PureComponent{
                 this.setState({genderStyle: 2})
                 break
             default:
-                this.setState({target: this.state.target += 1})
+            this.setState(prevState => ({
+              [selection]: prevState[selection] + 1
+              }))
                 break
         }
-        this.setDisplay()
+        const total = this.state.surveyPrompts + 1
+        this.setState({surveyPrompts: total})
+        this.setDisplay(total)
       }
     
       filterSurveyDisplay = () => {
         if (this.state.genderStyle === null) {
             return genderPreferences
         } else if (this.state.genderStyle === 0) {
-          //  Returning non-existent data - Non-binary survey page won't show
-          const nonBinary = this.state.surveyData.filter(data => data.type === this.state.display)
-          return [nonBinary[0], nonBinary[3], nonBinary[5]]
+            const nonBinary = this.state.surveyData.filter(data => data.type === this.state.display)
+            return [nonBinary[0], nonBinary[3], nonBinary[5]]
         } else {
-          return this.state.surveyData.filter(data =>
-            data.gender === this.state.genderStyle && data.type === this.state.display
-          )
+            const result = this.state.surveyData.filter(data =>
+              data.gender === this.state.genderStyle && data.type === this.state.display
+            )
+            return result
+        }
+      }
+
+      determineUserStyle = () => {
+        const total = this.state.Casual + this.state.Edgy + this.state.Streetwear
+        if(total === 4){
+          const styles = [this.state.Casual, this.state.Edgy, this.state.Streetwear]
+          const topSelected = styles.sort((a, b) => b - a)[0]
+          const result = Object.keys(this.state).find(e => {
+            if(!e.includes('genderStyle')){
+             return this.state[e] === topSelected
+            }
+          })
+          this.setState({result: result})
         }
       }
     
       onSubmit() {
-      //  axios post
-      // res.action set user style in store
-      // conditional check (if crateId exists subscribe)
+        //AXIOS.POST(USER STYLE)
```

Comment:
Add conditional to check if user was directed from a crate subscription click vs. profile survey selection.

Create two separate posts for each instance.